packages/webapp/src/pages/VerifyEmail.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Loading } from '@geist-ui/core';
+import { toast } from 'react-toastify';
+
+import DefaultLayout from '../layout/DefaultLayout';
+
+export default function VerifyEmail() {
+    const [serverErrorMessage, setServerErrorMessage] = useState('');
+    const [email, setEmail] = useState('');
+    const [loaded, setLoaded] = useState(false);
+    const navigate = useNavigate();
+
+    const { uuid } = useParams();
+
+    useEffect(() => {
+        if (!uuid) {
+            navigate('/');
+        }
+    }, [uuid, navigate]);
+
+    useEffect(() => {
+        const getEmail = async () => {
+            const res = await fetch(`/api/v1/account/email/${uuid}`);
+            const response = await res.json();
+
+            if (res?.status === 200) {
+                const { email, verified } = response;
+
+                if (verified) {
+                    setServerErrorMessage('Email already verified. Please navigate to the login page');
```

Comment:
this could just redirect imo

Reply:
Yeah, updated with https://github.com/NangoHQ/nango/pull/2173/commits/b4654576c1c63665bdd96435a276a41067347ebe